macrosAkka/src/main/scala/com/softwaremill/macwire/macwireakka/internals/Crimper.scala

```diff
@@ -0,0 +1,86 @@
+package com.softwaremill.macwire.macwireakka.internals
+
+import akka.actor.{ActorRef, ActorRefFactory, Props}
+import com.softwaremill.macwire.Logger
+import com.softwaremill.macwire.dependencyLookup.DependencyResolver
+
+import scala.reflect.macros.blackbox
+
+private[macwire] final class Crimper[C <: blackbox.Context, T: C#WeakTypeTag](val c: C, log: Logger) {
+  import c.universe._
+  lazy val targetType: Type = implicitly[c.WeakTypeTag[T]].tpe
+  lazy val dependencyResolver = new DependencyResolver[c.type](c, log)
+
+  lazy val targetConstructor: Symbol = log.withBlock(s"Looking for targetConstructor for $targetType"){
+    val publicConstructors = targetType.members
+      .filter(m => m.isMethod && m.asMethod.isConstructor && m.isPublic)
+      .filterNot(isPhantomConstructor)
+    log.withBlock(s"There are ${publicConstructors.size} eligible constructors" ) { publicConstructors.foreach(c => log(showConstructor(c))) }
+    val isInjectAnnotation = (a: Annotation) => a.toString == "javax.inject.Inject"
```

Comment:
Sure! This will require modularize and refactor a lill bit of the 'macros' project. I can do it :)